tests/azure_iot_hub_client_ut.c

```diff
@@ -0,0 +1,351 @@
+    assert_int_equal( AzureIoTHubClient_Init( pxTestIoTHubClient,
+                                              ucHostname, sizeof( ucHostname ) - 1,
+                                              ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                              &xHubClientOptions,
+                                              ucBuffer,
+                                              sizeof( ucBuffer ),
+                                              prvGetUnixTime,
+                                              &xTransportInterface ),
+                      eAzureIoTHubClientSuccess );
+}
+
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTHubClient_Init_Failure( void ** state )
+{
+    AzureIoTHubClient_t xTestIoTHubClient;
+    AzureIoTHubClientOptions_t xHubClientOptions = { 0 };
+
+    ( void ) state;
+
+    /* Fail init when null hostname passed */
+    assert_int_not_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                                  NULL, 0,
+                                                  ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                                  &xHubClientOptions,
+                                                  ucBuffer,
+                                                  sizeof( ucBuffer ),
+                                                  prvGetUnixTime,
+                                                  &xTransportInterface ),
+                          eAzureIoTHubClientSuccess );
+
+    /* Fail init when null deviceId passed */
+    assert_int_not_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                                  ucHostname, sizeof( ucHostname ) - 1,
+                                                  NULL, 0,
+                                                  &xHubClientOptions,
+                                                  ucBuffer,
+                                                  sizeof( ucBuffer ),
+                                                  prvGetUnixTime,
+                                                  &xTransportInterface ),
+                          eAzureIoTHubClientSuccess );
+
+    /* Fail init when null buffer passed */
+    assert_int_not_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
+                                                  ucHostname, sizeof( ucHostname ) - 1,
+                                                  ucDeviceId, sizeof( ucDeviceId ) - 1,
+                                                  &xHubClientOptions,
+                                                  NULL, 0,
+                                                  prvGetUnixTime,
+                                                  &xTransportInterface ),
+                          eAzureIoTHubClientSuccess );
+
+    /* Fail init when null AzureIoTGetCurrentTimeFunc_t passed */
+    assert_int_not_equal( AzureIoTHubClient_Init( &xTestIoTHubClient,
```

Comment:
>assert_int_not_equal [](start = 4, length = 20)

lets check for equal for exact behaviour we our checking

Reply:
yea I was going to say the same thing. done